scripts/yarn/vets-website-assets.js

```diff
@@ -0,0 +1,171 @@
+/* eslint-disable no-param-reassign */
+/* eslint-disable no-console */
+const fetch = require('cross-fetch')
+const fs = require('fs-extra')
+const path = require('path')
+
+/**
+ * This script gathers additional assets for building the front-end of VA.gov.
+ *
+ * Most of the assets are in the form of compiled javascript or css files.
+ * Additional assets are sourced directly from the vets-website repo.
+ *
+ * In the case of BUILD_TYPE=localhost, all assets are symlinked directly from a local vets-website repo
+ * instead of requesting anything from a bucket.
+ */
+
+const prodBucket = 'https://prod-va-gov-assets.s3-us-gov-west-1.amazonaws.com'
+const stagingBucket =
+  'https://staging-va-gov-assets.s3-us-gov-west-1.amazonaws.com'
+const devBucket = 'https://dev-va-gov-assets.s3-us-gov-west-1.amazonaws.com'
+const localBucket = path.resolve(
+  __dirname,
+  '../../../vets-website/build/localhost/generated'
+)
+
+// Available bucket options, default is the vagovprod bucket.
+const BUILD_TYPE_BUCKET = {
+  localhost: localBucket,
+  vagovdev: devBucket,
+  vagovstaging: stagingBucket,
+  vagovprod: prodBucket,
+}
+
+// This file exists in the above buckets and contains a list to all compiled assets that should be included.
+const fileManifestPath = 'generated/file-manifest.json'
+
+// Path to assets in a vets-website repo cloned adjacent to next-build.
+const vetsWebsiteAssetPath = '../vets-website/src/site/assets'
+
+const destinationPath = './public/generated/'
+
+// Function that loops through to download all compiled js assets listed in a bucket's manifest.
+async function downloadFromLiveBucket(buildtype) {
+  const bucket = BUILD_TYPE_BUCKET[buildtype]
+
+  const fileManifestRequest = await fetch(`${bucket}/${fileManifestPath}`)
+  const fileManifest = await fileManifestRequest.json()
+
+  const files = []
+
+  files[fileManifestPath] = {
+    path: fileManifestPath,
+    contents: Buffer.from(JSON.stringify(fileManifest)),
+  }
```

Comment:
I'm not sure I understand what's going on with `files`. This initializes `files` as an array but then attempts to assign entries as an object (associative array). I don't think lines 51-54 will have any effect.

Then, below, we see something similar on lines 76-81. Again, I don't think this will have any effect, but I think the bigger point is that I don't see anything happening with `files` regardless. It's not returned, or used in any way, from what I can tell.

Reply:
I now realize some of this isn't actually new code (showing as added because of the changed file extension). I still would love to perhaps clean it up, but I'm not sure about scope here.

Reply:
It's a good question. fwiw I cribbed this from the content-build implementation https://github.com/department-of-veterans-affairs/content-build/blob/main/src/site/stages/build/plugins/download-assets.js#L21

`files` is not returned, and this can probably be cleaned up further. I ran with it as-is because it did download the assets as expected 🤷

Reply:
removed this, not needed. thank you.